torch/autograd/_functions/initializers.py

```diff
@@ -0,0 +1,20 @@
+from ..function import InplaceFunction
+
+
+class Zero(InplaceFunction):
+
+    @staticmethod
+    def forward(ctx, i, inplace=False):
+        if inplace:
+            ctx.mark_dirty(i)
+            result = i.zero_()
+        else:
+            result = i.new(i.size()).zero_()
```

Comment:
This doesn't work exactly like `zero_()`. It's more like `zero_().contiguous()`

Reply:
You can do expand, but only in the non-inplace case

Reply:
👍 updated this